front/app/component-library/components/Checkbox/index.tsx

```diff
@@ -176,7 +184,45 @@ const Checkbox = ({
         {label}
       </Box>
       {labelTooltipText && <IconTooltip content={labelTooltipText} />}
-    </StyledBox>
+    </Box>
+  ) : (
+    <Box
```

Comment:
If no label prop is passed it means we (should) provide a label ourself in the place where this component is rendered. We don't want to have two labels, so I made this split. We don't wrap with a label if no label prop is provided. To make the Checkbox still work on click, I had to add a new handler.

In a subsequent PR, this component will be fully split to make it much cleaner.